vendor/github.com/olivere/elastic/client.go

```diff
@@ -381,7 +381,7 @@ func SetHealthcheckInterval(interval time.Duration) ClientOptionFunc {
 
 // SetMaxRetries sets the maximum number of retries before giving up when
 // performing a HTTP request to Elasticsearch.
-func SetMaxRetries(maxRetries int) func(*Client) error {
+func SetMaxRetries(maxRetries int) ClientOptionFunc {
```

Comment:
another question, : )
the package `github.com/olivere/elastic/` we import is a third party lib we use.  If we modify the code here, and someone use the `godep`  to get the package from the origin location of github, the compile will fail.
so, the better way i think is taking a PR to `github.com/olivere/elastic/` to fix this.